### ISSUE TITLE
Support hard line break `\` on reflow (for mdformat compat)

### ISSUE THREAD

**Issue by ahgraber** [rvben/rumdl#110]
My current markdown lint and parse workflow uses markdownlint-cli and mdformat. I would like to use rumdl, but mdformat [enforces `\` as a hard line break](https://mdformat.readthedocs.io/en/stable/users/style.html#hard-line-breaks) rather than `  ` (space space). rumdl does not retain the spacing I'd expect on this (during reflow) (e.g., it does not retain a newline with list indentation.

e.g.,

```md
- [GPT-5 Pro](https://platform.openai.com/docs/models/gpt-5-pro) is available (and expensive!) at $15/$120 per million input/output tokens, maximum 272,000 output tokens.\
  OpenAI links to [Background mode](https://platform.openai.com/docs/guides/background), which allows devs to initiate longer-running tasks (GPT-5 Pro, Deep Research, etc.) without concern for timeouts through the use of a poll-able response object.
```

becomes

```md
- [GPT-5 Pro](https://platform.openai.com/docs/models/gpt-5-pro) is available (and expensive!) at $15/$120 per million input/output tokens, maximum 272,000 output tokens.\ OpenAI links to [Background mode](https://platform.openai.com/docs/guides/background), which allows devs to initiate longer-running tasks (GPT-5 Pro, Deep Research, etc.) without concern for timeouts through the use of a poll-able response object.
```

**Comment by rvben** [comment]
Hi @ahgraber,

Completely correct observation. I did not know '\\' can be used as a hard line break. I learned something today. I will add support in the upcoming version.
